src/main/java/unicash/model/commons/Amount.java

```diff
@@ -19,7 +21,7 @@ public class Amount {
      */
     public Amount(double amount) {
         checkArgument(isValidAmount(amount), MESSAGE_CONSTRAINTS);
-        this.amount = amount;
+        this.amount = Math.round(amount * 100.0) / 100.0;
```

Comment:
Do we want to immediately truncate all Amount values as we store them? An alternative would be to still store as the value inputted but only alter the UI to display 2dp. I think that might be better instead of modifying what the user inputted

Reply:
Else, could add some documentation in the class to state that the Amount only stores 2dp values and automatically truncates

Reply:
I'll reply to this before the other ones cus it's important - the reason is, if we only change the values at the UI level, there are a lot of discrepancies that arise and those are only just the ones I noticed yesterday.

For example, editing a transaction with a non-2dp amount would reflect the 2dp value in the TransactionCard, but the ResultDisplay box would still show the exact value, and that's because that information comes from UniCashMessage's formatTransaction method. Another example would be budgetting or balance calculating - any testers/users might have issues when their balance or budget is calculated when fringe amounts or too precise amounts are entered. e.g. A budget set at $200 might never be internally mathematically flagged as reached if the user inputs 199.999999 but any UI components depending on that will be modified and flagged to have been exceeded.

Therefore, I strongly suggest that we truncate all amounts to a maximum of 2 decimal places from the very point of instantiation so that no other discrepancies like this can arise later on, and we can completely avoid any possible bugs. Additionally, I think it also suits our target user profile as University students won't be dealing with any transactions that would require a higher than standard currency precision (like trading/currency exchange I guess? I'm actually not too sure abt transactions more precise than 2dp irl)

Reply:
Sure, I think that sounds good, just have to add in some proper documentation of this in the class and our DG.